### DISCUSSION TITLE
What is the preferred way to share code between iOS and Mac?

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#7581] [Q&A]
We are using DI for platform specific functionality, but often want to use the same code for iOS and Mac. Where should this code be located? Currently we are using the Platforms folder, but that would mean we duplicate code for iOS and Mac when really it can be shared.

We could put the code in a base class above Platforms with ifdefs for iOS and MAC but that feels... unclean. Is that really the right way?

**Comment by xtuzy** [comment]
you can custom define Condition at .csproj let you r folder or files compile when target is ios and mac. i feel use #if __IOS__ #endif more easy🤪
